### PR TITLE
♻️ Refactor strict token logic

### PR DESCRIPTION
**Pull request by MarioBatalha**

### CODE REVIEW

**Review by atilasousa — APPROVED** [review]

**Review by KipandaJr — APPROVED** [review]

File: views/components/select-token-modal/select-token-modal-body.tsx [hunk 1]
```diff
@@ -64,38 +65,42 @@ const SelectTokenModalBody: FC<SelectTokenModalBodyProps> = ({
...
-      isSearchAddress &&
-      (faucet
-        ? FAUCET_COINS.map(({ type }) => type)
-        : STRICT_TOKENS_TYPE[network as Network]
-      ).includes(search))
+    data &&
+    ((!isSearchAddress && filterSelected === TokenOrigin.Strict) ||
+      (filterSelected === TokenOrigin.Strict &&
+        isSearchAddress &&
+        data.strictTokensType.includes(search)))
```
Comment (KipandaJr) [inline views/components/select-token-modal/select-token-modal-body.tsx:74]: Duplicated condition

File: views/components/select-token-modal/select-token-modal-body.tsx [hunk 2]
```diff
@@ -64,38 +65,42 @@ const SelectTokenModalBody: FC<SelectTokenModalBodyProps> = ({
...
-        ]}
-      />
-    );
+    if (
+      data &&
+      ((!isSearchAddress && filterSelected === TokenOrigin.Strict) ||
+        (filterSelected === TokenOrigin.Strict &&
+          isSearchAddress &&
+          data.strictTokensType.includes(search)))
+    )
```
Comment (KipandaJr) [inline views/components/select-token-modal/select-token-modal-body.tsx:82]: duplicated condition

File: utils/coin/index.ts [hunk 3]
```diff
@@ -78,21 +77,19 @@ const coinMetadataToToken = (coin: CoinMetadataWithType): Token => ({
...
   new Promise((resolve) => {
     if (
-      STRICT_TOKENS_TYPE[network].includes(type) ??
+      strictTokens?.strictTokensType.includes(type) ??
       WORMHOLE_TOKENS_TYPE[network].includes(type) ??
       SUI_BRIDGE_TOKENS_TYPE[network].includes(type)
     )
       return resolve(
-        STRICT_TOKENS[network].find(
+        WORMHOLE_TOKENS[network].find(
```
Comment (KipandaJr) [inline utils/coin/index.ts:90]: why u remove STRICT_TOKEN and added WORMHOLE_TOKENS?

**Review by KipandaJr — CHANGES_REQUESTED** [review]

**Review by KipandaJr — CHANGES_REQUESTED** [review]
image

**Pull request closed without merging**